src/pybind/mgr/cephadm/module.py

```diff
@@ -3080,6 +3086,21 @@ def _get_prometheus_credentials(self) -> Tuple[str, str]:
             self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return (user, password)
 
+    @handle_orch_error
+    def generate_certificates(self, module_name: str) -> Optional[Dict[str, str]]:
+        import socket
+        supported_moduels = ['dashboard', 'prometheus']
+        if module_name not in supported_moduels:
+            raise OrchestratorError(f'Unsupported modlue {module_name}. Supported moduels are: {supported_moduels}')
+
+        host_fqdns = [socket.getfqdn(self.get_hostname())]
```

Comment:
Should we use that `_invneotry_get_fqdn` function you added for this? As a note, we should also have the host FQDNs in the `Inventory` class. It may be better at some point to swap to that over using `socket.getfqdn` from inside the container as the one in the inventory comes from running `socket.getfqdn()` directly on the hosts (see `get_fqdn` function in `src/cephadm/cephadmlib/net_utils.py` and it's usage in `src/cephadm/cephadmlib/host_facts.py`)

Reply:
I moved this functionality to the mgr module as I think it makes more sense to be there (dns resolution is generic). I agree on your proposal to get the fqdn from the inventory. Actually using kcli I already ran in some issues related to that as the name we get inside the container is different from the one we get inside.